### PR TITLE
Game stats

### PR DESCRIPTION
**Pull request by whitneykidd**
Created methods in Game class: 
- highest_total_score
- lowest_total_score
- percentage_home_wins
- percentage_visitor_wins
- percentage_ties
- count_of_games_by_season
- average_goals_per_game
- average_goals_by_season

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -16,9 +16,53 @@ def self.all
...
+  def self.count_of_games_by_season
+    games_by_season = @@all.group_by { |game| game.season }
+    games_by_season.transform_values { |games| games.length }
+  end
+
+  def self.average_goals_per_game
+    (all_scores.sum.to_f / @@all.length.to_f).round(2)
+  end
+
+  def self.average_goals_by_season
```
Comment (whitneykidd) [inline lib/game.rb:55]: This method could probably be broken down to include helper methods in the future

File: lib/game.rb [hunk 2]
```diff
@@ -16,9 +16,53 @@ def self.all
...
+    games_by_season = @@all.group_by { |game| game.season }
+    games_by_season.transform_values do |game|
+      game_scores = game.map { |game| game.away_goals + game.home_goals }
+      (game_scores.sum.to_f / game_scores.length.to_f).round(2)
+    end
+  end
+
   def initialize(game_details)
     @game_id = game_details[:game_id].to_i
-    @season = game_details[:season].to_i
```
Comment (dborski) [inline lib/game.rb:21]: Removed the `.to_i` because we want season names to be returned as strings and not integers.

**Review by dborski — APPROVED** [review]

**Comment by IamNorma** [comment]
Looks good so far

File: lib/game.rb [hunk 3]
```diff
@@ -28,6 +28,22 @@ def self.lowest_total_score
...
+  end
+
+  def self.percentage_visitor_wins
+    visitor_wins = @@all.find_all { |game| game.home_goals < game.away_goals}.count
+    ( visitor_wins.to_f / @@all.length.to_f ).round(1) * 100
+  end
+
+  def self.percentage_ties
+    ties = @@all.find_all { |game| game.home_goals == game.away_goals}.count
+    ( ties.to_f / @@all.length.to_f ).round(1) * 100
```
Comment (kiverso) [inline lib/game.rb:43]: lots of code shared between the home_wins, visitor_wins, and ties. Seems like a good opportunity for refactor once we get everything functional.

**Pull request merged**